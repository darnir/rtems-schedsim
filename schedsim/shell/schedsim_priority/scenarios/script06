echo "*** TEST 06 ***"
rtems_init
echo Create 0x0a010001 
task_create TA1 1
task_create TA2 2
task_create TA3 3
semaphore_create SEM1
semaphore_obtain SEM1 0
semaphore_obtain SEM1 0
semaphore_release SEM1
semaphore_delete SEM1
echo "*** END OF TEST 06 ***"
exit
# We will not get here
