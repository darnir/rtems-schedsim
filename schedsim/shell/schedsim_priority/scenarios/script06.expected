*** TEST 06 ***
  Thread Heir: 0x09010001 priority=255
  Thread Executing: 0x09010001 priority=255
Create 0x0a010001
Creating task with:  Preemption: yes Timeslicing: no
Task (TA1) created: id=0x0a010001, priority=1
Task (TA1) starting: id=0x0a010001, priority=1
  Thread Heir: 0x0a010001 priority=1
  Thread Executing: 0x0a010001 priority=1
Creating task with:  Preemption: yes Timeslicing: no
Task (TA2) created: id=0x0a010002, priority=2
Task (TA2) starting: id=0x0a010002, priority=2
Creating task with:  Preemption: yes Timeslicing: no
Task (TA3) created: id=0x0a010003, priority=3
Task (TA3) starting: id=0x0a010003, priority=3
Semaphore (SEM1) created: id=0x1a010001
Obtain semaphore (0x1a010001) with timeout 0
  Thread Heir: 0x0a010002 priority=2
  Thread Executing: 0x0a010002 priority=2
Obtain semaphore (0x1a010001) with timeout 0
  Thread Heir: 0x0a010003 priority=3
  Thread Executing: 0x0a010003 priority=3
Releasing semaphore (0x1a010001)
  Thread Heir: 0x0a010001 priority=1
  Thread Executing: 0x0a010001 priority=1
Deleting semaphore (0x1a010001)
*** END OF TEST 06 ***
